packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1539,46 +1553,51 @@ abstract class DiagnosticsNode {
     if (kReleaseMode) {
       return <String, Object>{};
     }
-    final bool hasChildren = getChildren().isNotEmpty;
-    return <String, Object>{
-      'description': toDescription(),
-      'type': runtimeType.toString(),
-      if (name != null)
-        'name': name,
-      if (!showSeparator)
-        'showSeparator': showSeparator,
-      if (level != DiagnosticLevel.info)
-        'level': describeEnum(level),
-      if (showName == false)
-        'showName': showName,
-      if (emptyBodyDescription != null)
-        'emptyBodyDescription': emptyBodyDescription,
-      if (style != DiagnosticsTreeStyle.sparse)
-        'style': describeEnum(style),
-      if (allowTruncate)
-        'allowTruncate': allowTruncate,
-      if (hasChildren)
-        'hasChildren': hasChildren,
-      if (linePrefix?.isNotEmpty == true)
-        'linePrefix': linePrefix,
-      if (!allowWrap)
-        'allowWrap': allowWrap,
-      if (allowNameWrap)
-        'allowNameWrap': allowNameWrap,
-      ...delegate.additionalNodeProperties(this),
-      if (delegate.includeProperties)
-        'properties': toJsonList(
-          delegate.filterProperties(getProperties(), this),
-          this,
-          delegate,
-        ),
-      if (delegate.subtreeDepth > 0)
-        'children': toJsonList(
-          delegate.filterChildren(getChildren(), this),
-          this,
-          delegate,
-        ),
-    };
+    Map<String, Object> result;
+    assert(() {
+      final bool hasChildren = getChildren().isNotEmpty;
+      result = <String, Object>{
+        'description': toDescription(),
+        'type': runtimeType.toString(),
+        if (name != null)
+          'name': name,
+        if (!showSeparator)
+          'showSeparator': showSeparator,
+        if (level != DiagnosticLevel.info)
+          'level': describeEnum(level),
+        if (showName == false)
+          'showName': showName,
+        if (emptyBodyDescription != null)
+          'emptyBodyDescription': emptyBodyDescription,
+        if (style != DiagnosticsTreeStyle.sparse)
+          'style': describeEnum(style),
+        if (allowTruncate)
+          'allowTruncate': allowTruncate,
+        if (hasChildren)
+          'hasChildren': hasChildren,
+        if (linePrefix?.isNotEmpty == true)
+          'linePrefix': linePrefix,
+        if (!allowWrap)
+          'allowWrap': allowWrap,
+        if (allowNameWrap)
+          'allowNameWrap': allowNameWrap,
+        ...delegate.additionalNodeProperties(this),
+        if (delegate.includeProperties)
+          'properties': toJsonList(
+            delegate.filterProperties(getProperties(), this),
+            this,
+            delegate,
+          ),
+        if (delegate.subtreeDepth > 0)
+          'children': toJsonList(
+            delegate.filterChildren(getChildren(), this),
+            this,
+            delegate,
+          ),
+      };
+      return true;
+    }());
+    return result;
```

Comment:
In release mode, this function returns an empty map (line 1554). In profile mode, this function now returns null. Is there a reason why those two have to be different?

Reply:
(This difference is also present in other functions changed by this PR).

Reply:
Done. Thanks. I only kept kRelease in instances where it was calling super.toString() , didn't want to call twice.